### PR TITLE
Refactoring to postpone heavy resource loading in Safari

### PR DESCRIPTION
**Pull request by kirlat**
With this  change we're eliminating a `ContentProcess` and moving a user UI interaction related part of it into a UI controller and a part related to communication with a background into a content script.

This allow to reduce amount of files that are different between Safari and webextension to the minimum. There are only two different files now: a content script and a messaging service implementation. The rest of the code is fully shared between Safari and webextension.

A code that is related to background interaction has been moved to a content script so that we'll be able to create and initialize an instance of UI controller directly from there as needed, per message received from background. That would allow a more careful use of resources and more flexible usage scenarios (i.e. when UI controller instance could be destroyed per background request).

Please let me know what do you think. Thanks!

### CODE REVIEW

**Comment by kirlat** [comment]
@irina060981 thanks for your comments. As discussion of architecture may become lengthy, I created a separate issue to track this: #130. Can you please write there about the following (and of course anything else you would like to :smile:), in addition to the two questions asked there:
* What functionality that are in the UI controller now should belong to a content script?
* How can we make a content script more modular? Right now we have a UI controller and a messenger service there. What other components should it have?

Refactoring is hard, especially when deciding how to do it, considering sometimes conflicting requirements of all actors. But I think it is necessary if we want to end up with better, more extendable architecture. I believe we can do it together! Thanks!

**Comment by irina060981** [comment]
@kirlat, I tried to answer all your questions in all issues that is relevant to refactoring . Hope they will be helpfull and could give some useful thoughts :)
Anyway I think your task is really hard - refactoring is a very difficult process and way to ideal could be infinite. So be free to handle with my thoughts that way that would be as much useful for your process as you decide to do :)

**Comment by kirlat** [comment]
@irina060981, yes those are great explanations, thank you! I will need to think about it for a while.

**Comment by kirlat** [comment]
@balmas, @irina060981! Here is an update to address issues mentioned in comments to alpheios-project/components#266. Please review. Thanks!

File: package.json [hunk 1]
```diff
@@ -24,7 +24,7 @@
   },
   "homepage": "https://github.com/alpheios-project/webextension#readme",
   "devDependencies": {
-    "alpheios-components": "github:alpheios-project/components",
+    "alpheios-components": "file:../components",
```
[the remark this answers is not in the document]
Comment (kirlat) [reply package.json]: Fixed in 407716670d4b634fe2303da76409aa37c5d49c64

**Comment by kirlat** [comment]
Since it's the whole Xcode project folder was moved and not structural parts of it, it had no effect on the Xcode projects. For me it was a matter of closing a project from the old directory and reopening it from a new one.

Files that are external to the project (CSS and JS files that needs to be loaded to the Safari) are set in Xcode as relative links (that's important and we should keep it this way, I believe). They are relative to, if I remember correctly (writing this from a non-mac machine), a user home directory. So if we move Xcode project files around, those links still stay true.

**Comment by irina060981** [comment]
I tried to upload, build this project (with components repo appropriate branch) - everything is working well!
The only moment - @kirlat, be carefull with merging - as I have already pulled the PR with changes is App folder and SafariApp folder (everything in Swift).

**Pull request closed without merging**